src/pages/contentful/entity-harvester/index.vue

```diff
@@ -0,0 +1,186 @@
+        }
+      },
+
+      entityParamsFromUrl(url) {
+        url = url.replace(new RegExp('^https?://api.europeana.eu/entity'), BASE_URL);
+        if (isEntityUri(url)) {
+          return entityParamsFromUri(url);
+        }
+        const pageMatch = url.match(new RegExp('^https?://[^/]+(/[a-z]{2})?/collections/(person|topic|time)/([0-9]+)+'));
+        if (pageMatch) {
+          const type = pageMatch[2];
+          const id = pageMatch[3];
+          return { type, id };
+        }
+        throw new Error;
+      },
+
+      showError(error) {
+        this.contentfulExtensionSdk.dialogs.openAlert({
+          title: 'Error',
+          message: error
+        });
+        this.message = 'Failed';
+      },
+
+      // TODO: set up a configurable map for other fields to avoid hard-coding them here
+      populateFields(response, id) {
+        const enPrefLabel = langMapValueForLocale(response.prefLabel, 'en', { omitAllUris: true }).values[0];
+        // set field values
+        this.entry.fields.identifier.setValue(response.id); // data.europeana.eu URI
+
+        if (this.entry.fields.slug) {
+          this.entry.fields.slug.setValue(getEntitySlug(id, enPrefLabel)); // slug
+        }
+
+        if (this.entry.fields.type) {
+          this.entry.fields.type.setValue(getEntityTypeHumanReadable(response.type)); // entity type
+        }
+
+        // set name field from `prefLabel`
+        if (this.entry.fields.name) {
+          this.entry.fields.name.removeValue();
+          if (response.prefLabel) {
+            this.entry.fields.name.setValue(enPrefLabel);
+          }
+        }
+
+        // set description from different fields based on entity type
+        if (this.entry.fields.description) {
+          this.entry.fields.description.removeValue();
+          this.entry.fields.description.setValue(this.entityDescriptionFromResponse(response));
+        }
+
+        // set image field from `isShownBy`
+        if (this.entry.fields.image) {
+          this.entry.fields.image.removeValue();
+          if (response.isShownBy) {
+            this.entry.fields.image.setValue(response.isShownBy.thumbnail);
+          }
+        }
+      },
+
+      entityDescriptionFromResponse(response) {
+        switch (response.type) {
+        case 'Agent':
+          // use `biographicalInformation`
+          // NB: this is in JSON-LD expanded form
+          return langMapValueForLocale(response.biographicalInformation, 'en').values[0];
+        case 'Concept':
+          // use `note`
+          // NB: language map with each value being an array of literals
+          return langMapValueForLocale(response.note, 'en').values[0];
+        case 'Organization':
+          // use `description`
+          // NB: language map with each value being a single literal
+          return langMapValueForLocale(response.description, 'en').values[0];
+        case 'Timespan':
+          // TODO: use what? `${response.begin} to ${response.end}`?
```

Comment:
Begin to end sounds a bit "obvious", let's keep it blank as you have right now, so an editor can add their own description? Or is there a more descriptive description available in the API?

Reply:
nothing I'm aware of/have seen on the examples I've checked.